### PR TITLE
(maint) Shutdown jruby executor

### PR DESCRIPTION
**Pull request by justinstoller**
This is the code at https://github.com/puppetlabs/puppetserver/pull/2316 + a test based on the JRuby test at https://github.com/jruby/jruby/pull/6127/files + plus fixing a broken require (it's probably best to go through this PR per commit because of that).

### CODE REVIEW

File: test/integration/puppetlabs/services/jruby/jruby_puppet_pool_int_test.clj [hunk 1]
```diff
@@ -618,7 +622,44 @@
            (jruby-protocol/flush-jruby-pool! jruby-service)
            ; wait until the flush is complete
            (await pool-agent)
-           (is (logged? #"In cleanup fn"))))))))
+           (is (logged? #"In cleanup fn")))))))
+  (testing "flushing the pool does not leaving orphaned timeout threads"
```
Comment (Magisus) [inline test/integration/puppetlabs/services/jruby/jruby_puppet_pool_int_test.clj:625]: "leaving" -> "leave"

**Comment by mwaggett** [comment]
lgtm, besides the typo Maggie mentioned

**Comment by justinstoller** [comment]
grrr, I fixed this yesterday and then forgot to push it up

File: test/integration/puppetlabs/services/jruby/jruby_puppet_pool_int_test.clj (hunk 1 above)
Comment (justinstoller) [reply test/integration/puppetlabs/services/jruby/jruby_puppet_pool_int_test.clj]: updated

**Comment by Magisus** [comment]
Is this relevant to 5.x? Hard to say from the JRuby issue if it's 9k specific...

**Comment by justinstoller** [comment]
Damn, good question. It's probably applicable in 5.x w/ 9k, though we'll need to test that it doesn't blow up in 1.7.

**Comment by justinstoller** [comment]
I'm not getting to the 1.7 testing anytime soon. I'm going to merge this and if we find the time we can backport it.

**Pull request merged**